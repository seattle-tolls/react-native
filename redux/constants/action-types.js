export const SET_TOLL_INFO = 'SET_TOLL_INFO'
export const FETCH_TOLL_INFO = 'FETCH_TOLL_INFO'

export const SET_TOLL_NAME = 'SET_TOLL_NAME'