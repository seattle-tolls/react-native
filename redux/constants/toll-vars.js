export const TOLL_99 = 'toll99'
export const TOLL_520 = 'toll520'