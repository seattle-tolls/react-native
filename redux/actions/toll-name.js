import { SWITCH_TOLL_NAME } from '../constants/action-types'

export const switchTollName = (payload) => ({
  type: SWITCH_TOLL_NAME, payload,
})