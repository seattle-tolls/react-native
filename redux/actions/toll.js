import { SET_TOLL_INFO, FETCH_TOLL_INFO } from '../constants/action-types'

export const setTollInfo = (payload) => ({
  type: SET_TOLL_INFO, payload,
})

export const fetchTollInfo = () => ({
  type: FETCH_TOLL_INFO,
})