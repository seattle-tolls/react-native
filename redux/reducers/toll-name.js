import { SWITCH_TOLL_NAME } from '../constants/action-types'
// import { createSelector } from 'reselect'

const initialState = ''

const tollNameReducer = (tollName = initialState, action) => {
  switch (action.type) {
    case SWITCH_TOLL_NAME:
      return action.payload
    default:
      return tollName
  }
}

// TODO: finish the get name selector
export const getName = state => {
  console.log('TOLLNAME-->', state.tollName)
}

export default tollNameReducer