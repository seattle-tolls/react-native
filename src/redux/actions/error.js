import { REPORT_ERROR } from '../constants/action-types'

export const reportError = (error) => ({
  type: REPORT_ERROR, error,
})