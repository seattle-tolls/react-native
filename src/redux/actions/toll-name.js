import { SET_TOLL_NAME } from '../constants/action-types'

export const setTollName = (payload) => ({
  type: SET_TOLL_NAME, payload,
})
