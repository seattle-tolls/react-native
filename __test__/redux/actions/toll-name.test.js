import { setTollName } from '../../../src/redux/actions/toll-name'
import { SET_TOLL_NAME } from '../../../src/redux/constants/action-types'

describe('ACTIONS', () => {
  describe('actions/toll-name', () => {

    const tollNames = ['toll99', 'toll520']
    tollNames.forEach(payload => {
      it(`should run setTollName on ${payload}`, () => {
        const tollName = setTollName(payload)
        expect(tollName.type).toBe(SET_TOLL_NAME)
        expect(tollName.payload).toEqual(payload)
        expect(tollName).toMatchSnapshot()
      })
    })
  })
})