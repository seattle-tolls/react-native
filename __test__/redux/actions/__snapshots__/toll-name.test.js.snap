// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ACTIONS actions/toll-name should run setTollName on toll99 1`] = `
Object {
  "payload": "toll99",
  "type": "SET_TOLL_NAME",
}
`;

exports[`ACTIONS actions/toll-name should run setTollName on toll520 1`] = `
Object {
  "payload": "toll520",
  "type": "SET_TOLL_NAME",
}
`;
